src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
I'm still not sure what you're proposing.
Today, we have this IOV struct used (as buffer) for transfer functions:
```
typedef struct ucp_dt_iov {
    void           *buffer; /**< Pointer to a data buffer */
    ucp_datatype_t dt;      /**< Type of the data buffer */
    size_t         count;   /**< Amount of items in the data buffer */
} ucp_dt_iov_t;
```

We also have a function for creating strided datatypes, which can by used (as datatype) for transfer functions AND for IOV specifications:
```
typedef struct ucp_dt_strided_dim {
    size_t extent;
    size_t count;
} ucp_dt_strided_dim_t;

ucp_datatype_t ucp_dt_make_strided(ucp_datatype_t datatype,
                                   ucp_dt_strided_dim_t* dimensions,
                                   unsigned dimension_count,
                                   unsigned interleave_ratio,
                                   int is_reusable);
```

- Could you please clarify what you mean by "strided type as the parent of IOV type" and "strided has a handle which can hold cache"?

Clarification: this proposal is for both IOV and strided can be created as "reusable"/"persistent" and hold caches as metadata.